src/lib/s3.test.js

```diff
@@ -38,173 +52,273 @@ const VCAP_SERVICES = {
+    });
+
+    it('returns an error if S3 is not configured', () => {
+      const url = getPresignedURL(Key, Bucket, null);
+      expect(url).toMatchObject({ url: null, error: new Error('S3 is not configured.') });
+    });
+
+    it('calls getSignedUrl() with correct parameters', () => {
+      const url = getPresignedURL(Key, Bucket, mockS3);
+      expect(url).toMatchObject({ url: 'https://example.com', error: null });
+      expect(mockGetURL).toHaveBeenCalled();
+      expect(mockGetURL).toHaveBeenCalledWith('getObject', { Bucket, Key, Expires: 360 });
+    });
+
+    it('calls getSignedUrl() with incorrect parameters', async () => {
+      mockGetURL.mockImplementationOnce(() => { throw fakeError; });
+      const url = getPresignedURL(Key, Bucket, mockS3);
+      expect(url).toMatchObject({ url: null, error: fakeError });
+      expect(mockGetURL).toHaveBeenCalled();
+      expect(mockGetURL).toHaveBeenCalledWith('getObject', { Bucket, Key, Expires: 360 });
+    });
   });
-});
 
-describe('s3Uploader.deleteFileFromJobS3', () => {
-  const Bucket = 'fakeBucket';
-  const Key = 'fakeKey';
-  const anotherFakeError = Error({ statusCode: 500 });
-  it('calls deleteFileFromS3Job() with correct parameters', async () => {
-    const mockDeleteObject = jest.spyOn(s3, 'deleteObject').mockImplementation(() => ({ promise: () => Promise.resolve({ status: 200, data: {} }) }));
-    const got = deleteFileFromS3Job({ data: { fileId: 1, fileKey: Key, bucket: Bucket } });
-    await expect(got).resolves.toStrictEqual({
-      status: 200, data: { fileId: 1, fileKey: Key, res: { data: {}, status: 200 } },
-    });
-    expect(mockDeleteObject).toHaveBeenCalledWith({ Bucket, Key });
+  describe('s3Uploader.deleteFileFromS3', () => {
+    const Bucket = 'fakeBucket';
+    const Key = 'fakeKey';
+    const anotherFakeError = Error('fake');
+    let mockDeleteObject;
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    beforeEach(() => {
+      mockS3.deleteObject = jest.fn();
+      mockS3.deleteObject.mockImplementation(() => ({ promise: () => Promise.resolve('good') }));
+    });
+
+    it('throws an error if S3 is not configured', async () => {
+      await expect(deleteFileFromS3(Key, Bucket, null)).rejects.toThrow('S3 is not configured.');
+    });
+
+    it('calls deleteFileFromS3() with correct parameters', async () => {
+      const got = deleteFileFromS3(Key, Bucket, mockS3);
+      await expect(got).resolves.toBe('good');
+      expect(mockS3.deleteObject).toHaveBeenCalledWith({ Bucket, Key });
+    });
+
+    it('throws an error if promise rejects', async () => {
+      mockS3.deleteObject.mockImplementation(
+        () => ({ promise: () => Promise.reject(anotherFakeError) }),
+      );
+      const got = deleteFileFromS3(Key, undefined, mockS3);
+      await expect(got).rejects.toBe(anotherFakeError);
+      expect(mockS3.deleteObject).toHaveBeenCalledWith({ Bucket, Key });
+    });
```

Comment:
Is there some reason why you are passing undefined as bucket name? it doesn't seem to be part of the test, and passing it as undefined means it defaults to the destructured value from the s3config function, so line 274 will not pass. I'm just not clear as to what we are trying to test here.